search/scorer/scorer_similarity.go

```diff
@@ -57,17 +61,35 @@ func (sqs *SimilarityQueryScorer) Score(ctx *search.SearchContext,
 	similarityMatch *index.VectorDoc) *search.DocumentMatch {
 	rv := ctx.DocumentMatchPool.Get()
 
-	rv.IndexInternalID = append(rv.IndexInternalID, similarityMatch.ID...)
+	if sqs.includeScore || sqs.options.Explain {
+		var scoreExplanation *search.Explanation
+		score := similarityMatch.Score
+
+		// if the query weight isn't 1, multiply
+		if sqs.queryWeight != 1.0 {
+			score = score * sqs.queryWeight
+		}
 
-	// TODO Need to replace dummy score with actual score obtained from vector doc.
+		if sqs.includeScore {
+			rv.Score = score
+		}
 
-	if sqs.includeScore {
-		rv.Score = float64(rand.Intn(50))
+		if sqs.options.Explain {
+			rv.Expl = scoreExplanation
+		}
 	}
 
+	rv.IndexInternalID = append(rv.IndexInternalID, similarityMatch.ID...)
 	return rv
 }
 
 func (sqs *SimilarityQueryScorer) Weight() float64 {
-	return 1
+	return sqs.queryBoost * sqs.queryBoost
+}
+
+func (sqs *SimilarityQueryScorer) SetQueryNorm(qnorm float64) {
+	sqs.queryNorm = qnorm
+
+	// update the query weight
+	sqs.queryWeight = sqs.queryBoost
```

Comment:
the queryWeight for a child searcher is normalized weights (with respect to other child searchers in a disjunction searcher, for eg). so it's better to do a 

`sqs.queryWeight = sqs.queryBoost * sqs.queryNorm`